addons/fleet/migrations/12.0.0.1/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,161 @@
+DEL fleet.service.type: fleet.type_service_service_10
+DEL fleet.service.type: fleet.type_service_service_11
+DEL fleet.service.type: fleet.type_service_service_12
+DEL fleet.service.type: fleet.type_service_service_13
+DEL fleet.service.type: fleet.type_service_service_14
+DEL fleet.service.type: fleet.type_service_service_15
+DEL fleet.service.type: fleet.type_service_service_16
+DEL fleet.service.type: fleet.type_service_service_17
+DEL fleet.service.type: fleet.type_service_service_18
+DEL fleet.service.type: fleet.type_service_service_19
+DEL fleet.service.type: fleet.type_service_service_2
+DEL fleet.service.type: fleet.type_service_service_3
+DEL fleet.service.type: fleet.type_service_service_5
+DEL fleet.service.type: fleet.type_service_service_6
+DEL fleet.service.type: fleet.type_service_service_7
+DEL fleet.service.type: fleet.type_service_service_8
+DEL fleet.service.type: fleet.type_service_service_9
+DEL fleet.vehicle.model: fleet.model_a1
+DEL fleet.vehicle.model: fleet.model_a3
+DEL fleet.vehicle.model: fleet.model_a4
+DEL fleet.vehicle.model: fleet.model_a5
+DEL fleet.vehicle.model: fleet.model_a6
+DEL fleet.vehicle.model: fleet.model_a7
+DEL fleet.vehicle.model: fleet.model_a8
+DEL fleet.vehicle.model: fleet.model_agila
+DEL fleet.vehicle.model: fleet.model_ampera
+DEL fleet.vehicle.model: fleet.model_antara
+DEL fleet.vehicle.model: fleet.model_astra
+DEL fleet.vehicle.model: fleet.model_astragtc
+DEL fleet.vehicle.model: fleet.model_classa
+DEL fleet.vehicle.model: fleet.model_classb
+DEL fleet.vehicle.model: fleet.model_classc
+DEL fleet.vehicle.model: fleet.model_classcl
+DEL fleet.vehicle.model: fleet.model_classcls
+DEL fleet.vehicle.model: fleet.model_classe
+DEL fleet.vehicle.model: fleet.model_classgl
+DEL fleet.vehicle.model: fleet.model_classglk
+DEL fleet.vehicle.model: fleet.model_classm
+DEL fleet.vehicle.model: fleet.model_classr
+DEL fleet.vehicle.model: fleet.model_classs
+DEL fleet.vehicle.model: fleet.model_classslk
+DEL fleet.vehicle.model: fleet.model_classsls
+DEL fleet.vehicle.model: fleet.model_combotour
+DEL fleet.vehicle.model: fleet.model_corsa
+DEL fleet.vehicle.model: fleet.model_insignia
+DEL fleet.vehicle.model: fleet.model_meriva
+DEL fleet.vehicle.model: fleet.model_mokka
+DEL fleet.vehicle.model: fleet.model_q3
+DEL fleet.vehicle.model: fleet.model_q5
+DEL fleet.vehicle.model: fleet.model_q7
+DEL fleet.vehicle.model: fleet.model_serie1
+DEL fleet.vehicle.model: fleet.model_serie3
+DEL fleet.vehicle.model: fleet.model_serie5
+DEL fleet.vehicle.model: fleet.model_serie6
+DEL fleet.vehicle.model: fleet.model_serie7
+DEL fleet.vehicle.model: fleet.model_seriehybrid
+DEL fleet.vehicle.model: fleet.model_seriem
+DEL fleet.vehicle.model: fleet.model_seriex
+DEL fleet.vehicle.model: fleet.model_seriez4
+DEL fleet.vehicle.model: fleet.model_tt
+DEL fleet.vehicle.model: fleet.model_zafira
+DEL fleet.vehicle.model: fleet.model_zafiratourer
+# Keep those data, do not remove it
```

Comment:
Yes, my customer were using those values.
Technically I could do what you are suggesting, yes. But I don't think is a correct approach: here we are migrating data and those data were existing in the customer's database for years.
Why do you think the unused values are provoking noise? Just because Odoo moved them to demo data? Doesn't sound a good reason to me.

Reply:
Yes, it is a good reason. OpenUpgrade should preserve at maximum previous data, but trying to be in line with new features / data layout. That's why you should try to remove them, and if not possible, then keep them.

Reply:
> trying to be in line with new features / data layout

This is a better reason :+1:

Reply:
Pedro, I have some problems when using the `delete_records_safely_by_xml_id` in the pre-migration script. It seems failing because the fleet models are not yet loaded in the registry at the moment the function executed. Not sure if I'm doing something wrong.
I would keep the scripts as they are, since I don't have time to continue to develop them. They are working fine by the way.

Reply:
Yeah, they should be done in post-migration. You can do:

- Mark them as noupdate="1", so they are not removed.
- Try to remove them on post. If not possible, remove the ir.model.data record.

Reply:
With `noupdate="1" ` and moving the scripts in post-migration it works better, but still the `delete_records_safely_by_xml_id` raises a not-null exception on the `model_id` field. I could silent the exception and continue to add tricks for going further. But I'm still not fully convinced of this approach.

For example, what about any eventual custom module that adds a not-mandatory many2one field to `fleet.vehicle.model`? In case the to-be-deleted records are used there, no raise will be performed and that information will be lost. 

I prefer to keep the scripts as they are. If someone really wants to remove those records, I would say: go ahead and close this PR.

Reply:
OK, let's move on and let it this way, as always manual deletion can be done.